doc/man3/BIO_sendmmsg.pod

```diff
@@ -0,0 +1,167 @@
+operating system. This occurs when a packet is received which was sent by the
+local system. This occurs regardless of whether the packet's destination address
+was the loopback address or the IP address of a local non-loopback interface.
+This is an OS-specific limitation.
+
+The I<stride> argument must be set to C<sizeof(BIO_MSG)>. This argument
+facilitates backwards compatibility if fields are added to B<BIO_MSG>. Callers
+must zero-initialize B<BIO_MSG>.
+
+I<num_msg> should be sent to the maximum number of messages to send or receive,
+which is also the length of the array pointed to by I<msg>.
+
+Unlike most BIO functions, these functions explicitly support multi-threaded
+use. Multiple concurrent writers and multiple concurrent readers are permitted
+in any combination. As such, these functions do not clear, set, or otherwise
+modify BIO retry flags. The return value must be used to determine whether an
+operation should be retried; see below.
+
+BIO_dgram_set_local_addr_enable() and BIO_dgram_get_local_addr_enable() control
+whether local address support is enabled. To enable local address support, call
+BIO_dgram_set_local_addr_enable() with an argument of 1. The call will fail if
+local address support is not available for the platform.
+BIO_dgram_get_local_addr_enable() retrieves the value set by
+BIO_dgram_set_local_addr_enable().
+
+BIO_dgram_get_local_addr_cap() determines if the B<BIO> is capable of supporting
+local addresses.
+
+BIO_IS_ERRNO() and BIO_UNPACK_ERRNO() are used with the return values of
+BIO_sendmmsg() and BIO_recvmmsg(); see below.
+
+=head1 NOTES
+
+Some implementations of the BIO_sendmmsg() and BIO_recvmmsg() BIO methods might
+always process at most one message at a time, for example when OS-level
+functionality to transmit or receive multiple messages at a time is not
+available.
+
+=head1 RETURN VALUES
+
+BIO_sendmmsg() and BIO_recvmmsg() return the number of messages sent or
+received. Where a positive value n is returned, all entries in the B<BIO_MSG>
+array from 0 through n-1 inclusive have their I<data_len> and I<flags> fields
+updated with the results of the operation on that message. If the call was to
+BIO_recvmmsg() and the I<peer> or I<local> fields of that message are non-NULL,
+the B<BIO_ADDR> structures they point to are written with the relevant address.
+
+If an error occurs before any message has been sent or received, a negative
+value is returned. -2 is returned if the BIO_sendmmsg() or BIO_recvmmsg() method
+is not supported by the BIO; -1 is returned if a local has been specified or
+requested and local address support is not available or not enabled; otherwise,
+a packed system error code is returned. The macro BIO_IS_ERRNO() evaluates to 1
+if a BIO_sendmmsg() or BIO_recvmmsg() return value represents a system error
+code. BIO_UNPACK_ERRNO() should then be used to obtain the system error code
+(EWOULDBLOCK, etc.) from the return value.
```

Comment:
I wonder why we do this. In other BIO calls we just signal error and let the caller examine errno directly.